nmoscommon/mdns/mdnsEngine.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+import zeroconf
+from nmoscommon.logger import Logger
+from .dnsListener import DNSListener
+from .mdnsCallbackHandler import MDNSAdvertisementCallbackHandler
+from .mdnsRegistration import MDNSRegistration
+from .mdnsExceptions import ServiceAlreadyExistsException
+from .mdnsInterfaceController import MDNSInterfaceController
+from .mdnsRegistrationController import MDNSRegistrationController
+from .mdnsSubscriptionController import MDNSSubscriptionController
+
+
+class MDNSEngine(object):
+
+    def start(self):
+        self.subscriptionController = MDNSSubscriptionController()
+        self.registrationController = MDNSRegistrationController()
+        self.interfaceController = MDNSInterfaceController()
+        self.logger = Logger('mdns-engine')
+
+    def stop(self):
+        self.close()
+
+    def close(self):
+        self.subscriptionController.close()
+        self.registrationController.close()
+        self.interfaceController.close()
+
+    def run(self):
+        pass
+
+    def register(self, name, regtype, port, txtRecord="", callback=None, interfaceIps=None):
+        if not interfaceIps:
+            interfaceIps = []
+        interfaces = self.interfaceController.getInterfaces(interfaceIps)
+        registration = MDNSRegistration(interfaces, name, regtype, port, txtRecord)
+        callbackHandler = MDNSAdvertisementCallbackHandler(callback, registration)
+        try:
+            self.registrationController.addRegistration(registration)
+        except zeroconf.NonUniqueNameException:
+            callbackHandler.entryColission()
+        except ServiceAlreadyExistsException:
+            callbackHandler.entryColission()
+        except zeroconf.Error as e:
+            callbackHandler.entryFailed()
+        else:
+            callbackHandler.entryEstablisted()
+
+    def update(self, name, regtype, txtRecord=None):
+        self.unregister(name, regtype)
+        self.register(name, type, txtRecord=txtRecord)
```

Comment:
Where does 'type' come from here? Should it be regtype?

Reply:
Looks like this might also be missing a 'port' parameter?

Reply:
The port parameter wasn't in the original interface - I assume the main aim of this function is to be able to update the txt record

Reply:
Indeed, it just looked like the call to register would fail if the port was omitted.